apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx

```diff
@@ -253,14 +264,24 @@ const UpdateModal = ({
       <Text alignSelf="center" variant="h5Inter">
         {t("manager.modal.title", { productName: deviceModel.productName })}
       </Text>
-      {cancel ? (
+      {err ? (
+        <DeviceCancel
+          error={err}
+          shouldReloadManagerOnCloseIfUpdateRefused={
+            props.shouldReloadManagerOnCloseIfUpdateRefused
+          }
+          onDrawerClose={onDrawerClose}
+          onRetry={handleReset}
+          onSkip={onSkip}
+        />
+      ) : cancel ? (
         <Cancel onContinue={onRequestCancel} onCancel={onRequestClose} />
       ) : showDisclaimer ? (
         <Disclaimer onContinue={() => setShowDisclaimer(false)} t={t} firmware={firmware} />
       ) : (
         <FlowStepper.Indexed
           activeKey={stateStepId}
```

Comment:
Can you improve the structure of the code ? :) As you update this file 

Like have something like this : 
```ts
{err ? ( <DeviceCancel {...} /> ): null}
 {cancel ? (  <Cancel onContinue={onRequestCancel} onCancel={onRequestClose} />)  : null}     
```

And so on? :) It'll be readable i think

Reply:
Ternary should be ok with 2 options